core/docs/Changelog.md

```diff
@@ -2,27 +2,45 @@
 
 ## Unreleased
 
-* Arrays are now created unpinned by default, they were created pinned
-  earlier. During IO operations unpinned arrays are automatically copied
-  to pinned memory. When arrays are directly passed to IO operations
-  programmers can choose to create them pinned to avoid a copy.  To
-  create pinned arrays, use the internal APIs with the `pinned*` prefix.
-
-* Changed the signature of 'Streamly.Data.Stream.handle' to make the exception
-  handler monadic.
-* Rethrow the exception promptly in bracketIO.
-
 ### Breaking
 
 * `ParserK` in `Streamly.Data.ParserK` is not specialized to arrays anymore. To
   adapt to the new code, change `ParserK a m b` to `ParserK (Array a) m b` where
-  `Array` comes from `Streamly.Data.Array`.
+  `Array` comes from `Streamly.Data.Array`. This change also changed the
+  signatures of `parseChunks` and `parseBreakChunks`.
+* Changed the signature of 'Streamly.Data.Stream.handle' to make the
+  exception handler monadic.
+* Rethrow exceptions promptly in `bracketIO`.
+* `getIndex` signature changed in `Streamly.Data.MutArray` and
+  `Streamly.Data.MutArray.Generic`.
+
+### Internal Changes
+* Fold constructor has changed, added a `final` field to support
```

Comment:
We should remove `Fold.rmapM`. Can unintentionally create an improper extract function.
Actually, we'll almost always create an improper extract function.
We can make `rmapM :: (b -> c) -> (b -> m c) -> Fold m a b -> Fold m a c`.
Can always use `rmapM (const undefined) mapper fold`. If planning to not use scans.